src/lib/utils/detect-insights-client.ts

```diff
@@ -0,0 +1,6 @@
+export default function hasDetectedInsightsClient(): boolean {
+  return (
+    typeof window !== 'undefined' &&
+    Boolean(window && (window as any).AlgoliaAnalyticsObject)
```

Comment:
Why do you recheck `window` here? It's already checked on the line above.

```suggestion
    Boolean((window as any).AlgoliaAnalyticsObject)
```

Reply:
I was worried that `window === null` for example

Reply:
Is there a reason for writing this in a defensive programming way? This is usually not what we do in JavaScript and in this codebase.

Reply:
it's not about being defensive. My concerns when writing this were:
1. having clear error messages: if you don't do this you get `TypeError: Cannot read property 'AlgoliaAnalyticsObject' of null`, which doesn't say much about what happened. 
2. not deceiving the reader thinking such a line can't fail `typeof window !== 'undefined' && window. AlgoliaAnalyticsObject`:
  Maybe writing it as `typeof window !== 'undefined' && window !== null` would have been better. 
  Or we can decide the window global can't be null, which is a safe assumption to my knowledge.

Reply:
That's what I meant. `window` cannot be `null`, otherwise the contract is not respected and everything would break (outside of InstantSearch).

Reply:
ok, it's a fair assumption. I've removed it here: https://github.com/algolia/instantsearch.js/pull/4191/commits/0246b92043145e6821d7c4203bb78b84e15d3cb7